src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -271,13 +273,27 @@ export const handleGetPublicForm: ControllerHandler<
     case AuthType.SP:
     case AuthType.CP:
       return SpcpService.extractJwtPayloadFromRequest(authType, req.cookies)
-        .map(({ userName }) =>
-          res.json({
+        .map(({ userName, exp, iat, rememberMe }) => {
+          if (!exp) {
+            {
+              logger.error({
+                message: 'Invalid expiry time for cookie',
+                meta: logMeta,
+              })
+            }
+            return res.json({ form: publicForm, isIntranetUser })
+          }
+          return res.json({
             form: publicForm,
             isIntranetUser,
-            spcpSession: { userName },
-          }),
-        )
+            spcpSession: {
+              userName,
+              iat,
+              exp,
+              rememberMe,
+            },
```

Comment:
nit: instead of destructuring and restructuring everything, we can just keep this as one `spcpSession` object

Reply:
I thought about it earlier, and since the `spcpSession` object might potentially contain secrets (which is partly why we wanted to change to using httponly), we should be explicit about destructuring and returning the properties in the form object, but I guess in this case no harm. edited